docs/compass/03-02-labels.md

```diff
@@ -0,0 +1,155 @@
+                  "enum": ["Go", "Java", "C#"]
+              }
+          }"
+}) {...}
+
+
+setApplicationLabel(applicationID: "123", key: "supportedLanguages", value:["Go"]) {...}
+
+```
+
+### Edit LabelDefinitions
+
+You can edit LabelDefinitions using a mutation. When editing a LabelDefinition, make sure that all labels are compatible with the new definition. Otherwise, the mutation is rejected with a clear message that there are Applications or Runtimes that have an invalid label according to the new LabelDefinition. In such a case, you must either:
+- Remove incompatible labels from specific Applications and Runtimes, or
+- Remove the old LabelDefinition from all labels using cascading deletion.
+
+For example, let's assume we have the following LabelDefinition:
+
+```graphql
+ key:"supportedLanguages",
+  schema:{
+              "type": "array",
+              "items": {
+                  "type": "string",
+                  "enum": ["Go", "Java", "C#"]
+              }
+          }
+```
+
+If you want to add a new language to the list, provide such a mutation:
+
+```
+updateLabelDefinition(in: {
+                        key:"supportedLanguages",
+                        schema:{
+                                    "type": "array",
+                                    "items": {
+                                        "type": "string",
+                                        "enum": ["Go", "Java", "C#","ABAP"]
+                                    }
+                                }
+                      }) {...}
+```
+
+### Remove LabelDefinitions
+
+Use this mutation to remove a LabelDefinition:
+
+```graphql
+deleteLabelDefinition(key: String!, deleteRelatedLabels: Boolean=false): LabelDefinition
+
+```
+This mutation allows you to remove only definitions that are not used. If you want to delete a LabelDefinition with all values, set the **deleteRelatedLabels** parameter to `true`.
+
+### Search for objects using LabelFilters
+
+You can define a LabelFilter to list the objects according to their labels. To search for a given Application or Runtime, use this query:
+
+```graphql
+ applications(filter: [LabelFilter!], first: Int = 100, after: PageCursor):  ApplicationPage!
+```
+
+To search for all objects with a given label despite their values, use this query:
+
+```graphql
+query {
+  applications(filter:[{key:"scenarios"
+    }]) {
+    data {
+      name
+      labels
+    }
+    totalCount
+  }
+}
+```
+
+To search for all objects assigned to the `default` scenario, use this query:
+
+```graphql
+query {
+  applications(filter:[{key:"scenarios",
+    # This field is optional. If not provided, the query returns every object with the given label, regardless its value.
+    query:"$[*] ? (@ == \"DEFAULT\")"}]) {
+    data {
+      name
+      labels
+    }
+    totalCount
+  }
+}
+```
+
+In the **query** field, use only the limited SQL/JSON path expressions. The supported syntax is `$[*] ? (@ == "{VALUE}" )`.
```

Comment:
We should also include example for filtering by given key and value. @tgorgol has details how to do this.